controllers/ovndbcluster_controller.go

```diff
@@ -579,8 +571,122 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 			}
+			// Delete DNS records for deleted services/pods
+			dnsData := &infranetworkv1.DNSData{}
+			dnsName := fmt.Sprintf("%s-%s-%d", v1beta1.DNSPrefix, serviceName, i)
+			dnsNamespace := helper.GetBeforeObject().GetNamespace()
+			err = helper.GetClient().Get(ctx, types.NamespacedName{Name: dnsName, Namespace: dnsNamespace}, dnsData)
+			if err != nil && !k8s_errors.IsNotFound(err) {
+				Log.Info(fmt.Sprintf("Error while getting DNS object: %v", err))
+			}
+			err := helper.GetClient().Delete(ctx, dnsData)
+			if err != nil {
+				Log.Info(fmt.Sprintf("Error while cleaning up the DNS record %s: %v", dnsName, err))
+			}
+
+		}
+	}
+
+	podList, err = ovndbcluster.OVNDBPods(ctx, instance, helper, serviceLabels)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	// Assuming that the replicas is greater than 0 AND networkAttachment != ""
+	// dbAddress will contain ovsdbserver-(nb|sb).openstack.svc
+	// if replicas == 0 or networkAttachment == "" dbAddress will be empty.
+	// Meaning that Status.DBAddress is not ready.
+	dbAddress := ""
+
+	// DNSData is needed to provide DNS outside the cluster (edpm nodes) which needs to have
+	// spec.NetworkAttachment != ""
+	if instance.Spec.NetworkAttachment != "" {
+		for i, ovnPod := range podList.Items {
+			if i >= int(*(instance.Spec.Replicas)) {
+				break
+			}
+			var dnsName string
+			var dnsIP string
+			var hostnames []string
+			dnsName = v1beta1.DNSPrefix + "-" + ovnPod.Name
+			// Get Hostname
+			svc, err := service.GetServiceWithName(
+				ctx,
+				helper,
+				ovnPod.Name,
+				ovnPod.Namespace,
+			)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			hostname := svc.ObjectMeta.Annotations[infranetworkv1.AnnotationHostnameKey]
+			hostnames = append(hostnames, hostname)
+
+			// Get IP
+			net_stat, err := nad.GetNetworkStatusFromAnnotation(ovnPod.Annotations)
+			if err != nil {
+				Log.Info(fmt.Sprintf("Error while getting the NAD for pod %s: %v", ovnPod.Name, err))
+				return ctrl.Result{}, err
+			}
+			for _, v := range net_stat {
+				if strings.Contains(v.Name, instance.Spec.NetworkAttachment) {
+					// Return only IPv4 at the moment
+					if !strings.Contains(v.IPs[0], ":") {
+						dnsIP = v.IPs[0]
+						break
+					}
+				}
+			}
+
+			if len(dnsIP) == 0 {
+				Log.Info(fmt.Sprintf("Error while getting pod %s %s IP, IP is empty", ovnPod.Name, instance.Spec.NetworkAttachment))
+				continue
+			}
+
+			// Create DNSRecord
+			var DNSHosts []infranetworkv1.DNSHost
+			// ovsdbserver-(nb|sb)-n entry
+			DNSHost := infranetworkv1.DNSHost{}
+			DNSHost.IP = dnsIP
+			DNSHost.Hostnames = hostnames
+			// ovsdbserver-(sb|nb) entry
+			headlessDnsHostname := ovndbcluster.ServiceNameSB + "." + instance.Namespace + ".svc"
+			if instance.Spec.DBType == v1beta1.NBDBType {
+				headlessDnsHostname = ovndbcluster.ServiceNameNB + "." + instance.Namespace + ".svc"
+			}
+			dbAddress = fmt.Sprintf("tcp:%s:%d", headlessDnsHostname, svc.Spec.Ports[0].Port)
```

Comment:
So, since I need `svc.Spec.Ports[0].Port` I can't do it outside of loop (I could, but I'd need to call again `GetServiceWithName`). So  in order to avoid reconstructing this address each time what could I do is to check if dbAddress is empty, if so, set the value. This way we ensure to only set it once. 
Although this is adding another if (jump operation) into the code. 
I don't know what it's best:
- Comment explaining that's OK setting it up every iteration
- If to only set it once (with comment saying it's the same for all iterations)

Adding if, can be removed later if the other option is better